yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -247,26 +248,12 @@ export class AvmPersistableStateManager {
   /**
    * Get a contract's bytecode from the contracts DB, also trace the contract class and instance
    */
-  public async getBytecode(contractAddress: AztecAddress, selector: FunctionSelector): Promise<Buffer | undefined> {
+  public async getBytecode(contractAddress: AztecAddress, _selector: FunctionSelector): Promise<Buffer | undefined> {
     let exists = true;
-    // If the bytecode is not found, we let the executor decide that to do
-    const bytecode = await this.worldStateDB.getBytecode(contractAddress, selector);
     let contractInstance = await this.worldStateDB.getContractInstance(contractAddress);
-    // If the contract instance is not found, we assume it has not be deployed. We will also be unable to find the
-    // contract class as we will not have the id. While the class might exist, we hopefully won't need it to generate a proof (tbd).
-    if (contractInstance === undefined) {
```

Comment:
This is removed in favor of simply throwing an error, `this.worldStateDB.getBytecode` already throws if the contract instance is not found so this branch would never actually get called

Reply:
Hmm this gets a bit weird.... 

For nested calls, I think the CALL instruction should just return success=0. When it tries to create a new instance of the AvmSimulator, that will call getBytecode. If getBytecode throws an exception, it should be caught in the CALL instruction's execute function, which then skips `simulator.execute()` and creates `AvmContractCallResults` with empty output & reverted=true.

The `getBytecode` still needs to be traced because the AVM needs to prove that it doesn't exist!

It also gets a bit weirder now that we'll be proving multiple enqueued calls in the AVM. We need to be able to gracefully handle an enqueued call to a contract address that doesn't exist. So, even if an enqueued call is made to a contract that doesn't exist, we need trace its `getBytecode` so that the AVM knows to prove its non-existence!

Reply:
This is maybe irrelevant until recoverable reverts work. So maybe we can add this to the chunk of work for that @fcarreiro

Reply:
@dbanks12  sorry i actually changed this while you were reviewing it based on your previous comment.

What i've done for now is just hinted `exists = false`. The rest of the contract instance hints will be default values that are disregarded by the circuit